logstash-core/spec/logstash/agent_spec.rb

```diff
@@ -73,9 +73,9 @@
       #let(:agent_args) { { :logger => logger, :auto_reload => false, :reload_interval => 0.01, :config_path => config_file } }
       context "if state is clean" do
         it "should periodically reload_state" do
-          allow(subject).to receive(:clean_state?).and_return(false)
           expect(subject).to receive(:reload_state!).at_least(3).times
-          t = Thread.new { subject.execute }
+          t = Thread.new(subject) {|subject| subject.execute }
+          sleep 0.01 until (subject.running_pipelines? && subject.pipelines.values.first.ready?)
```

Comment:
+1 that they're not needed, but since it's a long line I'd rather keep them to help readability.